### PR TITLE
Add RadioButton

### PR DESCRIPTION
**Pull request by JoonghyunCho**
### Description of Change ###

Addition of RadioButton Control to Xamarin.Forms.
A RadioButton is a two-states button that can be either checked or unchecked.
A RadioButton cannot be unchecked by the user once checked.

### Bugs Fixed ###

N/A

### API Changes ###

Added:

New RadioButton class which has the following properties and events:
 - string Text { get; set; }
Gets or sets the Text displayed as the content of the radio button. This is a bindable property.
 - Color TextColor { get; set; }
Gets or sets the Color for the text of the radio button. This is a bindable property.
 - Font Font { get; set; }
Gets or sets the Font for the radio button text. This is a bindable property.
 - string FontFamily { get; set; }
Gets or sets the font family to which the font for the radio button text belongs. This is a bindable property.
 - double FontSize { get; set; }
Gets or sets the size of the font of the radio button text. This is a bindable property.
 - FontAttributes FontAttributes { get; set; }
Gets or sets a value that indicates whether the font for the radio button text is bold, italic, or neither. This is a bindable property.
 - bool IsChecked { get; set; }
Gets or sets a value that indicates whether the radio button is checked or unchecked. This is a bindable property.
 - string GroupName { get; set; }
Gets or sets the GroupName that is used to create a multiple-exclusion scope for a set of radio buttons. This is a bindable property.

 - event EventHandler Checked
	Occurs when the RadioButton is checked.
 - event EventHandler Unchecked
	Occurs when the RadioButton is unchecked.

### Behavioral Changes ###

N/A

### PR Checklist ###

- [ ] Has tests (if omitted, state reason in description)
- [x] Rebased on top of master at time of PR
- [x] Changes adhere to coding standard
- [x] Consolidate commits as makes sense

### Notes ###
 - Basic verification tests are done. The UI tests will be submitted soon.
 - Android renderer is completed. Renderers for other platforms will be submitted soon.

### CODE REVIEW

File: Xamarin.Forms.Core/RadioButton.cs [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+{
+    [RenderWith(typeof(_RadioButtonRenderer))]
+    public class RadioButton : View, IFontElement, ICheckedController
+    {
+        public static readonly BindableProperty TextProperty = BindableProperty.Create("Text", typeof(string), typeof(RadioButton), null,
+          propertyChanged: (bindable, oldVal, newVal) => ((RadioButton)bindable).InvalidateMeasureInternal(InvalidationTrigger.MeasureChanged));
+
+        public static readonly BindableProperty TextColorProperty = BindableProperty.Create("TextColor", typeof(Color), typeof(RadioButton), Color.Default);
+
+        public static readonly BindableProperty FontProperty = BindableProperty.Create("Font", typeof(Font), typeof(RadioButton), default(Font), propertyChanged: FontStructPropertyChanged);
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Core/RadioButton.cs]: In most controls, we obsoleted the Font property in favor of family, size, attribute. I would not add this.

File: Xamarin.Forms.Core/RadioButton.cs [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+        public static readonly BindableProperty FontFamilyProperty = BindableProperty.Create("FontFamily", typeof(string), typeof(RadioButton), default(string), propertyChanged: SpecificFontPropertyChanged);
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Core/RadioButton.cs]: please name the handlers `On[Property]PropertyChanged`

File: Xamarin.Forms.Core/RadioButton.cs [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+          defaultValueCreator: bindable => Device.GetNamedSize(NamedSize.Default, (RadioButton)bindable));
+
+        public static readonly BindableProperty FontAttributesProperty = BindableProperty.Create("FontAttributes", typeof(FontAttributes), typeof(RadioButton), FontAttributes.None,
+          propertyChanged: SpecificFontPropertyChanged);
+
+        public static readonly BindableProperty IsCheckedProperty = BindableProperty.Create("IsChecked", typeof(bool), typeof(RadioButton), false);
+
+        public static readonly BindableProperty GroupNameProperty = BindableProperty.Create("GroupName", typeof(string), typeof(RadioButton), null);
+
+        public Font Font
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Core/RadioButton.cs]: this can go

File: Xamarin.Forms.Core/RadioButton.cs [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+        public static readonly BindableProperty FontFamilyProperty = BindableProperty.Create("FontFamily", typeof(string), typeof(RadioButton), default(string), propertyChanged: SpecificFontPropertyChanged);
+
+        public static readonly BindableProperty FontSizeProperty = BindableProperty.Create("FontSize", typeof(double), typeof(RadioButton), -1.0, propertyChanged: SpecificFontPropertyChanged,
+          defaultValueCreator: bindable => Device.GetNamedSize(NamedSize.Default, (RadioButton)bindable));
+
+        public static readonly BindableProperty FontAttributesProperty = BindableProperty.Create("FontAttributes", typeof(FontAttributes), typeof(RadioButton), FontAttributes.None,
+          propertyChanged: SpecificFontPropertyChanged);
+
+        public static readonly BindableProperty IsCheckedProperty = BindableProperty.Create("IsChecked", typeof(bool), typeof(RadioButton), false);
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Core/RadioButton.cs]: no handler for IsChecked changed ?

File: Xamarin.Forms.Core/RadioButton.cs [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+
+        public static readonly BindableProperty GroupNameProperty = BindableProperty.Create("GroupName", typeof(string), typeof(RadioButton), null);
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Core/RadioButton.cs]: no handler for GroupName changed ?

File: Xamarin.Forms.Core/RadioButton.cs [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+        {
+            get { return (string)GetValue(GroupNameProperty); }
+            set { SetValue(GroupNameProperty, value ?? string.Empty); }
+        }
+
+        public event EventHandler Checked;
+
+        public event EventHandler Unchecked;
+
+        static void FontStructPropertyChanged(BindableObject bindable, object oldValue, object newValue)
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Core/RadioButton.cs]: this can go away

File: Xamarin.Forms.Platform.Android/Renderers/RadioButtonRenderer.cs [hunk 7]
```diff
@@ -0,0 +1,244 @@
...
+        public RadioButtonRenderer()
+        {
+        }
+
+        void CompoundButton.IOnCheckedChangeListener.OnCheckedChanged(CompoundButton buttonView, bool isChecked)
+        {
+            ((IViewController)Element).SetValueFromRenderer(RadioButton.IsCheckedProperty, isChecked);
+
+            if (isChecked)
+                ((ICheckedController)Element).SendChecked();
```
Comment (rmarinho) [inline Xamarin.Forms.Platform.Android/Renderers/RadioButtonRenderer.cs]: this should just be core handling ..

**Comment by rmarinho** [comment]
Closing for now, please reopen when feedback is address, and also you will need to rebase.

**Pull request closed without merging**